src/pubsub/ua_pubsub_writer.c

```diff
@@ -97,6 +99,107 @@ UA_DataSetWriterConfig_clear(UA_DataSetWriterConfig *pdsConfig) {
     memset(pdsConfig, 0, sizeof(UA_DataSetWriterConfig));
 }
 
+UA_StatusCode
+UA_Server_enableDataSetWriter(UA_Server *server, const UA_NodeId dataSetWriterIdent) {
+    UA_LOCK(&server->serviceMutex);
+    UA_StatusCode ret = UA_STATUSCODE_BADNOTFOUND;
+    UA_DataSetWriter *dataSetWriter = UA_DataSetWriter_findDSWbyId(server, dataSetWriterIdent);
+
+    if(dataSetWriter)
+    {
+        UA_NodeId wgId = dataSetWriter->linkedWriterGroup;
+        UA_WriterGroup* writerGroup = UA_WriterGroup_findWGbyId(server, wgId);
+        if(writerGroup->state == UA_PUBSUBSTATE_OPERATIONAL || writerGroup->state == UA_PUBSUBSTATE_PREOPERATIONAL) {
+            ret = UA_DataSetWriter_setPubSubState(server, dataSetWriter, UA_PUBSUBSTATE_PREOPERATIONAL,
+                                                    UA_STATUSCODE_GOOD);
+            if(ret != UA_STATUSCODE_GOOD) {
+                UA_LOG_ERROR_WRITER(&server->config.logger, dataSetWriter,
+                                        "Enable DataSetWriter failed - setPubSubState failed");
+            }
+        }
+        else
+        {
+            ret = UA_DataSetWriter_setPubSubState(server, dataSetWriter, UA_PUBSUBSTATE_PAUSED,
+                                                  UA_STATUSCODE_GOOD);
+        }
+    }
+    UA_UNLOCK(&server->serviceMutex);
+    return ret;
+}
+
+static UA_StatusCode
+startPubSubStateTimer(UA_Server *server, UA_DataSetWriter *dsw) {
+    UA_StatusCode ret = UA_STATUSCODE_BAD;
```

Comment:
Why do we need to wait 10ms to set the Writer operational?
Which change do we expect in the meantime and why can’t we set the new state immediately?
At the very least this should be documented in a comment.

Reply:
Currently there is no change expected and the writer can go immediately to operational state. I added this as a future considerations when there is some checks to be performed.

I would add a comment to make this clear